node/node.go

```
@@ -0,0 +1,452 @@
+		return nil, nil, err
+	}
+	w := wallet.NewSingleAddressWallet(walletKey, ws)
+
+	dbDir := filepath.Join(dir, "db")
+	if err := os.MkdirAll(dbDir, 0700); err != nil {
+		return nil, nil, err
+	}
+	dbConn := stores.NewSQLiteConnection(filepath.Join(dbDir, "db.sqlite"))
+
+	hostdbDir := filepath.Join(dir, "hostdb")
+	if err := os.MkdirAll(hostdbDir, 0700); err != nil {
+		return nil, nil, err
+	}
+	sqlStore, ccid, err := stores.NewSQLStore(dbConn, true)
+	if err != nil {
+		return nil, nil, err
+	} else if err := cm.ConsensusSetSubscribe(sqlStore, ccid, nil); err != nil {
+		return nil, nil, err
+	}
+
+	contractsDir := filepath.Join(dir, "contracts")
+	if err := os.MkdirAll(contractsDir, 0700); err != nil {
+		return nil, nil, err
+	}
+
+	objectsDir := filepath.Join(dir, "objects")
+	if err := os.MkdirAll(objectsDir, 0700); err != nil {
+		return nil, nil, err
+	}
+
+	cleanup := func() error {
+		errs := []error{
+			g.Close(),
+			cm.Close(),
+			tp.Close(),
+			sqlStore.Close(),
+		}
+		for _, err := range errs {
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	b := bus.New(syncer{g, tp}, chainManager{cm}, txpool{tp}, w, sqlStore, sqlStore, sqlStore, sqlStore)
+	return b, cleanup, nil
+}
+
+func newWorker(cfg WorkerConfig, walletKey consensus.PrivateKey) (*worker.Worker, func() error, error) {
+	b := bus.NewClient(cfg.BusAddr, cfg.BusPassword)
+	workerKey := blake2b.Sum256(append([]byte("worker"), walletKey...))
+	w := worker.New(workerKey, b)
+	return w, func() error { return nil }, nil
+}
+
+func newAutopilot(cfg AutopilotConfig, dir string) (*autopilot.Autopilot, func() error, error) {
+	store, err := stores.NewJSONAutopilotStore(dir)
+	if err != nil {
+		return nil, nil, err
+	}
+	b := bus.NewClient(cfg.BusAddr, cfg.BusPassword)
+	w := worker.NewClient(cfg.WorkerAddr, cfg.WorkerPassword)
+	a, err := autopilot.New(store, b, w, cfg.LoopInterval)
+	if err != nil {
+		return nil, nil, err
+	}
+	cleanup := func() error {
```

`autopilot` 's `Stop` should just return an error and you should return `a.Stop` here